### PR TITLE
Include UID in temp dir name.

### PR DESCRIPTION
**Pull request by lkmorlan**
This addresses #276.

### CODE REVIEW

File: lib/utils.js [hunk 1]
```diff
@@ -13,9 +13,13 @@ exports.emitErr = function (stream, err) {
 // and all matched files. Switching options does not break Sass cache so we do
 // it ourselves. Possibly a bug: https://github.com/sass/sass/issues/1830
 exports.createIntermediatePath = function (sources, matches, options) {
+	defaults.tempSubDir = 'gulp-ruby-sass';
```
Comment (robwierzbowski) [inline lib/utils.js]: I believe `defaults` is not available here. How about making the default `tempDir` in index.js contain the uid? Then we won't have to change this file or the clear function at all.

File: lib/utils.js [hunk 2]
```diff
@@ -13,9 +13,13 @@ exports.emitErr = function (stream, err) {
...
+	if (typeof process.getuid === 'function') {
+		defaults.tempSubDir += '-' + process.getuid();
+	}
```
Comment (robwierzbowski) [inline lib/utils.js]: When would `getuid` not be available?

Comment (robwierzbowski) [inline lib/utils.js]: Ah, not on Windows :P. Carry on :)

File: lib/utils.js (hunk 1 above)
Comment (lkmorlan) [inline lib/utils.js]: I thought the default tempDir was /tmp on POSIX. We can't change that name.

Comment (robwierzbowski) [inline lib/utils.js]: The `defaults` object is part of this plugin: https://github.com/lkmorlan/gulp-ruby-sass/blob/master/index.js#L22-L27

```js
var uid = typeof process.getuid === 'function' ? process.getuid() : '';

var defaults = {
	tempDir: osTmpdir() + uid,
	...
```

Comment (lkmorlan) [inline lib/utils.js]: If I understand correctly, osTmpdir() would normally be "/tmp", it would have to be:
```javascript
osTmpdir() + '/' + uid
```

Comment (robwierzbowski) [inline lib/utils.js]: True true. Go for it.

Comment (robwierzbowski) [inline lib/utils.js]: My concern is that we don't ask people to pass extra arguments to the clear cache function. Although that shouldn't be used often.

Also be sure to check tests when you're done.

Comment (robwierzbowski) [inline lib/utils.js]: Oh, please use `path.join` instead of `+`.

**Comment by lkmorlan** [comment]
I have made another commit which I think fixes the identified issues. Note that it changes the meaning of the tempDir configuration because gulp-ruby-sass will no longer be added to the end.

**Comment by robwierzbowski** [comment]
Love it. Thanks!

**Comment by robwierzbowski** [comment]
I'll ping this PR when I publish a new version.

**Comment by lkmorlan** [comment]
Thanks!

**Comment by stevenbower** [comment]
any chance of getting a build with this any time soon?

**Comment by robwierzbowski** [comment]
Publishing now, should be up in a minute.

**Pull request merged**